profiles/tests/test_profiles_graphql_api.py

```diff
@@ -1754,6 +1754,138 @@ def test_staff_user_can_filter_berth_profiles_by_addresses(rf, user_gql_client):
+    AddressFactory(profile=profile_2, postal_code="00100")
+    AddressFactory(profile=profile_3, postal_code="00200")
+
+    cat = SubscriptionTypeCategoryFactory(code="TEST-CATEGORY-1")
+    type_1 = SubscriptionTypeFactory(subscription_type_category=cat, code="TEST-1")
+    type_2 = SubscriptionTypeFactory(subscription_type_category=cat, code="TEST-2")
+
+    Subscription.objects.create(
+        profile=profile_1, subscription_type=type_1, enabled=True
+    )
+    Subscription.objects.create(
+        profile=profile_1, subscription_type=type_2, enabled=True
+    )
+
+    Subscription.objects.create(
+        profile=profile_2, subscription_type=type_1, enabled=True
+    )
+    Subscription.objects.create(
+        profile=profile_2, subscription_type=type_2, enabled=False
+    )
+
+    Subscription.objects.create(
+        profile=profile_3, subscription_type=type_1, enabled=True
+    )
+
+    service = ServiceFactory()
+    ServiceConnectionFactory(profile=profile_1, service=service)
+    ServiceConnectionFactory(profile=profile_2, service=service)
+    ServiceConnectionFactory(profile=profile_3, service=service)
+    group = GroupFactory()
+    user = user_gql_client.user
+    user.groups.add(group)
+    assign_perm("can_view_profiles", group, service)
+    request = rf.post("/graphql")
+    request.user = user
+
+    query = """
+        query getBerthProfiles($serviceType: ServiceType!, $subscriptionType: String, $postalCode: String){
+            profiles(
+                serviceType: $serviceType,
+                enabledSubscriptions: $subscriptionType,
+                addresses_PostalCode: $postalCode
+            ) {
+                edges {
+                    node {
+                        emails {
+                            edges {
+                                node {
+                                    email
+                                }
+                            }
+                        }
+                        phones {
+                            edges {
+                                node {
+                                    phone
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    """
+
+    # test for type 1 + postal code 00100
+
+    executed = user_gql_client.execute(
+        query,
+        variables={
+            "serviceType": ServiceType.BERTH.name,
+            "subscriptionType": type_1.code,
+            "postalCode": "00100",
+        },
+        context=request,
+    )
+    assert dict(executed["data"]) == generate_expected_data([profile_1, profile_2])
+
+    # test for type 2 + any postal code
+
+    executed = user_gql_client.execute(
+        query,
+        variables={
+            "serviceType": ServiceType.BERTH.name,
+            "subscriptionType": type_2.code,
+            "postalCode": "00",
```

Comment:
this passes the filter because all profiles have addresses that contain `00` in them, i.e. typical Helsinki addresses? in reality FE client can just skip passing any value to `addresses_PostalCode` filter to actually get any postal code, right?

Reply:
Yes that's by no means necessary filter. The example use case was just filtering by subscription and postal code. I'll remove the latter postal code filter since it's a bit misleading 🙂

Reply:
Actually, I'm not going to remove it because it was all about filtering by subscription **and** postal code. I added one more profile and tweaked the expected results a bit.

Reply:
yes, that works too 👍